hw12-solid/src/main/java/ru/otus/l12/homework/exception/MinCheckException.java

```diff
@@ -0,0 +1,7 @@
+package ru.otus.l12.homework.exception;
+
+public class MinCheckException extends RuntimeException {
+    public MinCheckException(String message) {
+        super(message);
```

Comment:
Сообщения в исключения, когда вы делаете свои исключения, лучше "упаковать" в само исключение, а не передавать его снаружи.

```suggestion
        super("Not enough cash for operation");
```

Можно и вашим путем пойти - хранить сообщения в константах (только enum для этого странно выглядит):
```
public MinCheckException() {
    super(ErrorsEnum.ERR_MIN_CHECK.getMessage());
}
```
При таком подходе, где бы вы не выбросили исключение, не надо думать о сообщении и не получится, что сообщение передали не то.

Еще лучше передать больше контекста в исключение:
```
public MinCheckException(int requestedSum) {
    super("Needed sum is less than min banknote. RequestedSum = " + requestedSum);
}
```

Reply:
Спасибо, поправила